tools/source_updater/updater/source.py

```diff
@@ -0,0 +1,164 @@
+
+    def upsert_secrets(self, source_name: str, configs: Set[Config]) -> None:
+        """
+        If config_path points to a directory, all json files from this repository will be uploaded. Else, only the file will be uploaded.
+        Note that this method will overwrite the content of <source-name>/secrets/updated_configurations since this is how ci_credentials
+        works.
+        """
+        gsm_credentials = json.loads(os.environ['GCP_GSM_CREDENTIALS'])
+        if not gsm_credentials:
+            raise ValueError("GSM credentials should be set as described in https://github.com/airbytehq/airbyte/blob/master/tools/ci_credentials/README.md")
+
+        self._set_up_config_files(source_name, configs)
+
+        secret_manager = SecretsManager(
+            connector_name=source_name,
+            gsm_credentials=gsm_credentials,
+        )
+        secrets = secret_manager.read_from_gsm()
+        secret_manager.update_secrets(secrets)
+
+    def _set_up_config_files(self, source_name: str, configs: Set[Config]):
+        """
+        In order of the current version of SecretManager to update secrets, updated secrets needs to be part of secrets/updated_configurations/*.json
+        """
+        secrets_folder_path = os.path.join(self._path_from_name(source_name), "secrets")
+        updated_config_path = os.path.join(secrets_folder_path, "updated_configurations")
+        if not os.path.exists(secrets_folder_path):
+            os.mkdir(secrets_folder_path)
+        if not os.path.exists(updated_config_path):
+            os.mkdir(updated_config_path)
+
+        for config in configs:
+            with open(os.path.join(updated_config_path, f"{config.name}.json"), 'w') as destination_file:
+                json.dump(config.content, destination_file)
+
+    @staticmethod
+    def _copy_file(source_file_path: str, destination_file_path: str):
+        with open(source_file_path, 'rb') as source, open(destination_file_path, 'wb') as destination:
+            destination.write(source.read())
+
+    def _catalog_path(self, source_name: str) -> str:
+        return os.path.join(self._path_from_name(source_name), "integration_tests", "configured_catalog.json")
+
+    def _manifest_path(self, source_name: str) -> str:
+        return os.path.join(self._path_from_name(source_name), source_name.replace("-", "_"), "manifest.yaml")
+
+    def _path_from_name(self, source_name: str) -> str:
+        return os.path.join("airbyte-integrations", "connectors", source_name)
+
+
+import yaml.emitter
+import yaml.serializer
+import yaml.representer
+import yaml.resolver
+
+
+class _IndentingEmitter(yaml.emitter.Emitter):
+    def increase_indent(self, flow=False, indentless=False):
+        """Ensure that lists items are always indented."""
+        return super().increase_indent(
+            flow=False,
+            indentless=False,
+        )
+
+
+class _PrettyDumper(
```

Comment:
This is to reduce the diff between our current formatting and the default yaml lib one. There are still some diffs that are not caused by an actual change but it feel very manageable